commons/all/src/main/java/org/infinispan/commons/dataconversion/MediaType.java

```diff
@@ -33,6 +36,22 @@
 @ProtoTypeId(ProtoStreamTypeIds.MEDIA_TYPE)
 @SerializeWith(value = MediaType.MediaTypeExternalizer.class)
 public final class MediaType {
+   private static final Pattern TREE_PATTERN;
+   private static final Pattern PARAMETER_PATTERN;
+
+   static {
+      // Adapted from https://stackoverflow.com/a/48046041/55870
+      // Extended to support "*" as a media type (as used by java.net.HttpURLConnection)
+      // More details at https://bugs.openjdk.java.net/browse/JDK-8163921
+      // and https://tools.ietf.org/html/rfc7231#section-3.1.1.1
+      String ows = "[ \t]*";
+      String token = "[0-9A-Za-z!#$%&'*+.^_`|~-]+";
+      String quotedString = "\"(?:[^\"\\\\]|\\\\.)*\"";
+      String tree = ows + "((" + token + ")/" + token + "|\\*)" + ows;
+      String parameter = ";" + ows + "(" + token + ")=(" + token + "|" + quotedString + ")" + ows;
+      TREE_PATTERN = Pattern.compile(tree);
+      PARAMETER_PATTERN = Pattern.compile(parameter);
+   }
```

Comment:
Is this faster than the current code on ```master```?

Reply:
I haven't ran any performance test after adding the regular expressions, but I expect it to be faster because it goes over the input string only one time and it does fewer allocations: the code on master allocates new strings with `substring()`, `split()`, and `toCharArray()`.

But it shouldn't matter for performance IMO, as most performance-sensitive applications should be using the id of `application/x-protostream` instead of a textual media type. The regex change was motivated mostly by the fact that I wanted something that follows the RFCs as close as possible, and I found an answer on StackOverflow doing it with regular expressions :)

Reply:
Parsing of ```MediaType``` does impact performance, as every REST call will send a text for every request

Reply:
> The regex change was motivated mostly by the fact that I wanted something that follows the RFCs as close as possible

Understood, but this PR is for small performance improvements, and if we don't know if it will improve, I suggest separating the MediaType changes to another JIRA

Reply:
You were right @gustavonalle, I wrote a JMH benchmark and the regular expression parsing was a bit slower than master:

I eventually made it faster by expanding the character ranges, but then I realized it's too big a change and it needs its own PR. I only left the `matchesAll` field that speeds up comparisons in this PR.